daemon/src/maker_cfd.rs

```diff
@@ -168,6 +175,21 @@ impl Actor {
         Ok(())
     }
 
+    fn get_settlement_proposal(&self, order_id: OrderId) -> Result<(SettlementProposal, TakerId)> {
+        let (update_proposal, taker_id) = self
+            .current_pending_proposals
+            .get(&order_id)
+            .context("have a proposal that is about to be accepted")?;
+
+        let proposal = match update_proposal {
+            UpdateCfdProposal::Settlement { proposal, .. } => proposal,
```

Comment:
Can we just store the `proposal` in the hashmap instead? Then I think we can avoid this boilerplate.

Reply:
that was the case before rollover support was added - shall we just have 2 separate hashmaps here instead and zip them only for the feed purposes? that would probably make it easier to see what's going on inside the cfd actors.

Reply:
@thomaseizinger there are basically 3 choices:
- this boilerplate
- have 2 hashmaps internally (one for each enum variant) and zip them in `send_pending_proposals()`
- manage hashmap per variant in Rocket (one for proposals, one for rollovers)

None of solutions is perfect. Happy to change this if we have a clear decision, preferably in a separate cleanup PR as it's not the core of this change.

Reply:
> that was the case before rollover support was added - shall we just have 2 separate hashmaps here instead and zip them only for the feed purposes? that would probably make it easier to see what's going on inside the cfd actors.

I don't quite understand. You are introducing this hashmap here, yeah? Why can you not design it exactly for what you need?

Even outside of this, I think we should have two data structures. If we can avoid error cases by modelling the problem differently, then I think that is the superior solution.

Happy for this to be tackled in a follow-up!